src/mpyl/steps/models.py

```diff
@@ -199,6 +199,12 @@ class Input:
     required_artifact: Optional[Artifact] = None
     dry_run: bool = False
 
+    def as_spec(self, typ: Type[ArtifactSpec]):
```

Comment:
Is `typ` the syntax used for python generics or could you also just name it something like `spec_type`?